precompiles/utils/macro/docs/precompile_macro.md

```diff
@@ -0,0 +1,198 @@
+# `#[precompile]` procedural macro.
+
+This procedural macro allows to simplify the implementation of an EVM precompile or precompile set
+using an `impl` block with annotations to automatically generate:
+- the implementation of the trait `Precompile` or `PrecompileSet` (exposed by the `fp_evm` crate)
+- parsing of the method parameters from Solidity encoding into Rust type, based on the `EvmData`
+  trait (exposed by the `precompile-utils` crate)
+- a test to ensure the types expressed in the Solidity signature match the Rust types in the
+  implementation.
+
+## How to use
+
+Define your precompile type and write an `impl` block that will contain the precompile methods
+implementation. This `impl` block can have type parameters and a `where` clause, which will be
+reused to generate the `Precompile`/`PrecompileSet` trait implementation and the enum representing
+each public function of precompile with its parsed arguments.
+
+```rust,ignore
+pub struct ExemplePrecompile<R, I>(PhantomData<(R,I)>);
+
+#[precomile_utils::precompile]
+impl<R, I> ExemplePrecompile<R, I>
+where
+    R: pallet_evm::Config
+{
+    #[precompile::public("example(uint32)")]
```

Comment:
```suggestion
    #[precompile::public("exemple(uint32)")]
```1

Reply:
It's not the correct english spelling but it matches the generated function in the example